sortinghat/core/db.py

```diff
@@ -290,6 +291,77 @@ def delete_organization(trxl, organization):
                        target=op_args['organization'])
 
 
+def add_group(trxl, organization, parent, group_name):
+    """Add a group to the database.
+
+    This function adds a new group to the database using `group_name` as its identifier.
+    The new group will be created as a subgroup of a parent, if parent is provided.
+    The new group will also be linked to the organization object in `organization`.
+
+    Values assigned to `group_name` cannot be `None` or empty.
+
+    As a result, the function returns a new `Group` object.
+
+    :param trxl: TransactionsLog object from the method calling this one
+    :param organization: links the new group to this organization object
+    :param parent: parent group of the new group
+    :param group_name: name of the group group
+
+    :returns: a new group
+
+    :raises ValueError: raised when `group_name` is `None` or an empty string;
+    """
+    # Setting operation arguments before they are modified
+    op_args = {
+        'organization': None if not organization else organization.name,
+        'group_name': group_name,
+        'parent': None if not parent else parent.name
+    }
+
+    validate_field('group_name', group_name)
+
+    try:
+        if not parent:
+            group = Group.add_root(name=group_name, organization=organization)
+        else:
+            group = parent.add_child(name=group_name, organization=organization)
+        group.save()
+    except django.db.utils.IntegrityError as exc:
+        _handle_integrity_error(Group, exc)
+
+    trxl.log_operation(op_type=Operation.OpType.ADD, entity_type='group',
+                       timestamp=datetime_utcnow(), args=op_args,
+                       target=op_args['group_name'])
+    return group
+
+
+def find_group(organization, group_name):
```

Comment:
Same as before. It should be:

```
def find_group(group_name, organization=None):
```

Reply:
This function is should go also where the other `find_*` functions are.

Reply:
Done 👍